src/pages/TeamSettings/Service-Accounts-Table.vue

```diff
@@ -0,0 +1,486 @@
+
+      <template #expanded-item="{item}">
+        <td :colspan="4">
+          <v-list dense class="mx-8">
+            <v-subheader>{{ item.firstName }}'s Keys</v-subheader>
+            <v-list-item
+              v-if="!keys.filter(key => key.user_id === item.id).length"
+              >{{ item.firstName }} currently has no keys.</v-list-item
+            >
+            <v-list-item
+              v-for="key in keys.filter(key => key.user_id === item.id)"
+              :key="key.id"
+            >
+              <v-list-item-title>{{ key.name }}</v-list-item-title>
+              <v-list-item-subtitle
+                >Created
+                {{
+                  key.created_at ? formDate(key.created_at) : ''
+                }}</v-list-item-subtitle
+              >
+              <v-list-item-subtitle>
+                {{
+                  key.expires
+                    ? `Expires ${formatTimeRelative(key.expires)}`
+                    : 'Never Expires'
+                }}</v-list-item-subtitle
+              >
+              <v-tooltip bottom>
+                <template #activator="{ on }">
+                  <v-btn
+                    text
+                    fab
+                    x-small
+                    color="error"
+                    v-on="on"
+                    @click="
+                      keyToDelete = key
+                      dialogRemoveKey = true
+                    "
+                  >
+                    <v-icon>delete</v-icon>
+                  </v-btn>
+                </template>
+                Remove API key
+              </v-tooltip>
+            </v-list-item>
+          </v-list>
+        </td>
+      </template>
+
+      <template v-if="isTenantAdmin" #item.create="{ item }">
+        <v-btn
+          small
+          color="primary"
+          class="white--text"
+          @click="
+            selectedUser = item
+            dialogCreateKey = true
+          "
+          ><v-icon left>
+            add
+          </v-icon>
+          Create API Key
+        </v-btn>
+      </template>
+
+      <!-- ACTIONS -->
+      <template v-if="isTenantAdmin" #item.actions="{ item }">
+        <v-tooltip bottom>
+          <template #activator="{ on }">
+            <v-btn
+              text
+              fab
+              x-small
+              color="error"
+              v-on="on"
+              @click="
+                selectedUser = item
+                dialogRemoveUser = true
+              "
+            >
+              <v-icon>delete</v-icon>
+            </v-btn>
+          </template>
+          Remove account
+        </v-tooltip>
+      </template>
+    </v-data-table>
+
+    <!-- CREATE KEY DIALOG -->
+    <ConfirmDialog
+      v-model="dialogCreateKey"
+      :dialog-props="{ 'max-width': '500' }"
+      :disabled="!newKeyFormFilled"
+      title="Create an API key"
+      confirm-text="Create"
+      @cancel="resetNewKey"
+      @confirm="
+        createAPIKey({
```

Comment:
When I click 'Create API Key' nothing happens.  This may be because of cloud access but be good to have some indication of what's happened - can we add loading state and error handling?

Reply:
added loading state, there's error handling in the method (`createAPIKey`) that I think should be sufficient? Interested in your thoughts there.

Reply:
Can we set an alert that shows the error?- Should be able to do that from the catch block in your createAPIKey method.

Reply:
only if I remember to import the alert component!